### PR TITLE
hostname: add recipe

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: recipes-debian/hostname/hostname_debian.bb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+DEBIAN_UNPACK_DIR = "${WORKDIR}/${PN}"
+DEBIAN_PATCH_TYPE = "quilt"
+DEBIAN_QUILT_PATCHES = ""
+
+do_compile() {
+    make
+}
+
+do_install() {
+    BASEDIR=${D} make install
```
Comment (masami256) [inline recipes-debian/hostname/hostname_debian.bb]: Use oe_runmake is preferred. such as ``` oe_runmake 'BASEDIR=${D}' install ```

File: recipes-debian/hostname/hostname_debian.bb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+LICENSE = "GPLv2"
+LIC_FILES_CHKSUM = "file://COPYRIGHT;md5=86dc5e6760b0845ece4d5be3a9d397d9"
+
+inherit debian-package
+require recipes-debian/sources/hostname.inc
+DEBIAN_UNPACK_DIR = "${WORKDIR}/${PN}"
+DEBIAN_PATCH_TYPE = "quilt"
+DEBIAN_QUILT_PATCHES = ""
+
+do_compile() {
```
Comment (masami256) [inline recipes-debian/hostname/hostname_debian.bb]: You can remove this function that oe_runmake runs make command .

**Comment by masami256** [comment]
When add hostname pacakge in rootfs by ``` IMAGE_INSTALL_append += "hostname" ``` then build core-image-minimal,  I got following errors.

```
update-alternatives: Error: not linking /home/masami/pr-test/build/tmp-glibc/work/qemuarm64-emlinux-linux/core-image-minimal/1.0-r0/rootfs/bin/hostname to /bin/busybox.nosuid since /home/masami/pr-test/build/tmp-glibc/work/qemuarm64-emlinux-linux/core-image-minimal/1.0-r0/rootfs/bin/hostname exists and is not a link

ERROR: core-image-minimal-1.0-r0 do_rootfs: Postinstall scriptlets of ['busybox'] have failed. If the intention is to defer them to first boot,
then please place them into pkg_postinst_ontarget_${PN} ().
Deferring to first boot via 'exit 1' is no longer supported.
Details of the failure are in /home/masami/pr-test/build/tmp-glibc/work/qemuarm64-emlinux-linux/core-image-minimal/1.0-r0/temp/log.do_rootfs.
ERROR: core-image-minimal-1.0-r0 do_rootfs: 
ERROR: core-image-minimal-1.0-r0 do_rootfs: Function failed: do_rootfs
ERROR: Logfile of failure stored in: /home/masami/pr-test/build/tmp-glibc/work/qemuarm64-emlinux-linux/core-image-minimal/1.0-r0/temp/log.do_rootfs.16108
ERROR: Task (/home/masami/pr-test/repos/poky/meta/recipes-core/images/core-image-minimal.bb:do_rootfs) failed with exit code '1'
NOTE: Tasks Summary: Attempted 2861 tasks of which 2858 didn't need to be rerun and 1 failed.
```

**Comment by yuwata** [comment]
Thank you for the review and testing. I've force-pushed a revised version. PTAL.

**Comment by hiraku-wfs** [comment]
@yuwata Busybox has hostname feature and it is included in rootfs by default.
Why is this necessary?

**Comment by yuwata** [comment]
Yes. But Busybox does not provide `domainname` or friends, does it?

**Comment by hiraku-wfs** [comment]
I investigated it.
Poky provides hostname, domainname and friends by net-tools recipe.
On the other hand, Debian provides them by special hostname package although hostname source code is included also in net-tools.
The meta-debian-extended already has net-tools recipe, so I think you don't need this hostname recipe.

**Comment by yuwata** [comment]
net-tools of meta-debian-extended does not provide hostname command.

**Comment by hiraku-wfs** [comment]
@yuwata Could you enable hostname in net-tools?

**Comment by yuwata** [comment]
OK. I will try that direction.

**Pull request closed without merging**